src/test/scala/is/hail/expr/ir/CompileSuite.scala

```diff
@@ -636,5 +612,8 @@ class CompileSuite {
     }
 
     testStructN(5)
+    testStructN(5000)
+
+    // FIXME: testStructN(20000) hits the Class file size limit!
```

Comment:
Jesus, that's not even that big!  We need to shard to multiple classes?!  Before we do that, we should look at our bytecode and see if we can make it a bit more streamlined.

Reply:
hmmm...
```
The RuntimeException("Class file too large!") is not thrown when the class file size is too large (and it would be really strange if your class file exceeds 2GiB anyway), but rather when the number of items in the constant pool exceeds 65534.
```
https://stackoverflow.com/questions/48120065/asm-class-file-too-large

Reply:
In particular, we're probably generating too many methods or fields or a combination that is greater than 65534. see: https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.4

Reply:
It's vaguely annoying that ASM doesn't provide a facility for textually depicting the class file even when it's "too large" or otherwise malformed.

Reply:
2 fields and 199 methods, in this particular case. :shrug: I'm going to take a look at what else is happening.